docs/reference/8-admin-app/technical-articles/updating-admin-app-22-for-ed-fi-odsapi-cloud-deployment-for-azure-suite-3.md

```diff
@@ -0,0 +1,314 @@
+for AdminAppName
+
+Make sure to create Admin App specific sql user before the deployment
+(AdminAppSqlUserName, AdminAppSqlPassword). Please refer **Step 5 on****Steps**
+**for new installation** section below.
+
+**Deploy-EdFiOdsAdminApp** Expand source
+
+```json
+$params = @{
+
+        ResourceGroupName = " Existing resource group name "
+
+        AdminAppName = "AdminApp-Latest"
+
+        AppInsightLocation = "South Central US"
+
+        ProductionApiUrl = " Existing ODS API url "
+
+        SQLServerHostname = " Existing SQL server url "
+
+        SQLServerUserName = "sql administrator username"
+
+        SQLServerPassword = ConvertTo-SecureString "sql administrator password"  -AsPlainText -Force
+
+  AdminAppSqlUserName = "admin app sql username"
+
+  AdminAppSqlPassword = ConvertTo-SecureString "admin app sql password" -AsPlainText -Force
+
+    }
+
+Upgrade-AdminApp> .\Deploy-EdFiOdsAdminApp.ps1 @params
+```
+
+**EncryptionKey:**
+
+Base64-encoded 256 bit key appropriate for use with AES encryption. This is an
+
+optional parameter.  If user wants to provide own value, then use following
+script to generate:
+
+**Generate AES Encryption Key** Expand source
+
+```ps1
+  $aes = [System.Security.Cryptography.Aes]::Create()
+  $aes.KeySize = 256
+  $aes.GenerateKey()
+  $EncryptionKey = [System.Convert]::ToBase64String($aes.Key)
+```
+
+:::info note:
+  If user is not providing Encryption key, then key will be generated
+  during deployment.
+:::
+
+4. Once the application deployment done, user will be prompted to confirm
+deleting and recreating Admin App specific database tables. If yes, then
+deployment process will delete existing Admin App specific tables and re-create
+them with latest table schemas on EdFi\_Admin database.
+
+5\. Data validations and update:
+
+:::info note:
+  This step can be ignored if it is new installation
+:::
+
+1. We can persist existing vendor applications, key and secrets created. Need to
+manually update OdsInstance\_OdsInstanceId column on dbo.Applications table to
+have default Ods instance id. Since, latest AdminApp needs association between
+dbo.Applications and dbo.OdsInstances tables to filter applications for selected
+instance. On SharedInstance mode will be having only one OdsInstance
+on dbo.OdsInstances table.
+
+**Update Applications** Expand source
+
+```sql
+  DECLARE @odsinstanceid INT
+  SELECT TOP 1 @odsinstanceid = OdsInstanceId FROM [dbo].[OdsInstances]
+  Update [EdFi_Admin].[dbo].[Applications] set OdsInstance_OdsInstanceId = @odsinstanceid where OdsInstance_OdsInstanceId is null
+```
+
+2\. Vendor applications created using older AdminApp have prefixed with
+“Production”. May need to manually remove that prefix.
+
+### Steps for new installation
+
+1.Please refer [Ed-Fi ODS/API Cloud Deployment for Azure (Suite
+3)](https://edfi.atlassian.net/wiki/spaces/EXCHANGE/pages/22487832))  for new
```

Comment:
New link: `/getting-started/edfi-exchange/technology/ed-fi-odsapi-cloud-deployment-for-azure-suite-3`